src/common/meta/src/ddl/create_table_template.rs

```diff
@@ -173,3 +179,55 @@ impl CreateRequestBuilder {
         request
     }
 }
+
+fn prepare_partition_expr(
+    region_id: RegionId,
+    partition_exprs: &HashMap<RegionNumber, String>,
+) -> Partition {
+    let expr = partition_exprs.get(&region_id.region_number()).cloned();
+    if expr.is_none() {
+        warn!("region {} has no partition expr", region_id);
+    }
```

Comment:
Will this be too verbose for normal tables without partition?

Reply:
In a table without a partition, the default region's expr will be an empty string rather than None. This is a defensive measure to identify cases where options are missing.